code/mission/import/xwingmissionparse.cpp

```diff
@@ -0,0 +1,743 @@
+		else
+			sprintf(sexp_buf, "( is-destroyed-delay 0 \"%s\" )", arrival_fg_name);
+		Mp = sexp_buf;
+		return get_sexp_main();
+	}
+
+	if (fg->arrivalEvent == XWMArrivalEvent::ae_afg_disabled)
+	{
+		if (check_wing)
+			sprintf(sexp_buf, "( fotg-is-wing-disabled \"%s\" )", arrival_fg_name);
+		else
+			sprintf(sexp_buf, "( fotg-is-ship-disabled \"%s\" )", arrival_fg_name);
+		Mp = sexp_buf;
+		return get_sexp_main();
+	}
+
+	if (fg->arrivalEvent == XWMArrivalEvent::ae_afg_identified)
+	{
+		if (check_wing)
+			sprintf(sexp_buf, "( fotg-is-wing-identified \"%s\" )", arrival_fg_name);
+		else
+			sprintf(sexp_buf, "( fotg-is-ship-identified \"%s\" )", arrival_fg_name);
+		Mp = sexp_buf;
+		return get_sexp_main();
+	}
+
+	return Locked_sexp_true;
+}
+
+int xwi_determine_anchor(const XWingMission *xwim, const XWMFlightGroup *fg)
+{
+	int mothership_number = fg->mothership;
+
+	if (mothership_number >= 0)
+	{
+		if (mothership_number < (int)xwim->flightgroups.size())
+			return get_parse_name_index(xwim->flightgroups[mothership_number].designation.c_str());
+		else
+			Warning(LOCATION, "Mothership number %d is out of range for Flight Group %s", mothership_number, fg->designation.c_str());
+	}
+
+	return -1;
+}
+
+const char *xwi_determine_formation(const XWMFlightGroup *fg)
+{
+	switch (fg->formation)
+	{
+		case XWMFormation::f_Vic:
+			return "Double Vic";
+		case XWMFormation::f_Finger_Four:
+			return "Finger Four";
+		case XWMFormation::f_Line_Astern:
+			return "Line Astern";
+		case XWMFormation::f_Line_Abreast:
+			return "Line Abreast";
+		case XWMFormation::f_Echelon_Right:
+			return "Echelon Right";
+		case XWMFormation::f_Echelon_Left:
+			return "Echelon Left";
+		case XWMFormation::f_Double_Astern:
+			return "Double Astern";
+		case XWMFormation::f_Diamond:
+			return "Diamond";
+		case XWMFormation::f_Stacked:
+			return "Stacked";
+		case XWMFormation::f_Spread:
+			return "Spread";
+		case XWMFormation::f_Hi_Lo:
+			return "Hi-Lo";
+		case XWMFormation::f_Spiral:
+			return "Spiral";
+	}
+
+	return nullptr;
+}
+
+const char *xwi_determine_base_ship_class(const XWMFlightGroup *fg)
+{
+	switch (fg->flightGroupType)
+	{
+		case XWMFlightGroupType::fg_X_Wing:
+			return "T-65c X-wing";
+		case XWMFlightGroupType::fg_Y_Wing:
+			return "BTL-A4 Y-wing";
+		case XWMFlightGroupType::fg_A_Wing:
+			return "RZ-1 A-wing";
+		case XWMFlightGroupType::fg_TIE_Fighter:
+			return "TIE/ln Fighter";
+		case XWMFlightGroupType::fg_TIE_Interceptor:
+			return "TIE/In Interceptor";
+		case XWMFlightGroupType::fg_TIE_Bomber:
+			return "TIE/sa Bomber";
+		case XWMFlightGroupType::fg_Gunboat:
+			return nullptr;
+		case XWMFlightGroupType::fg_Transport:
+			return "DX-9 Stormtrooper Transport";
+		case XWMFlightGroupType::fg_Shuttle:
+			return "Lambda-class T-4a Shuttle";
+		case XWMFlightGroupType::fg_Tug:
+			return nullptr;
+		case XWMFlightGroupType::fg_Container:
+			return nullptr;
```

Comment:
New addition that could be used here are `DV-3 Container` and `DV-3 Cargo Freighter#NoCargo`